code/main.py

```
@@ -0,0 +1,122 @@
+import pandas as pd
+import numpy as np
+from sentence_transformers import SentenceTransformer
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import Dense, Input, Embedding, Flatten, Concatenate
+from tensorflow.keras.utils import to_categorical
+import matplotlib.pyplot as plt
+
+# Step 1: Load and Merge Datasets
+def load_data():
```

Ruff (0.8.2)

Missing return type annotation for public function `load_data`

(ANN201)